OpenRA.Mods.RA/Effects/GpsDot.cs

```diff
@@ -60,19 +69,20 @@ public GpsDot(Actor self, GpsDotInfo info)
 			disguise = Exts.Lazy(() => self.TraitOrDefault<Disguise>());
 			cloak = Exts.Lazy(() => self.TraitOrDefault<Cloak>());
 
-			watcher = new Cache<Player, GpsWatcher>(p => p.PlayerActor.Trait<GpsWatcher>());
 			frozen = new Cache<Player, FrozenActorLayer>(p => p.PlayerActor.Trait<FrozenActorLayer>());
 
-			showToPlayer = self.World.Players.ToDictionary(x => x, x => false);
+			stateByPlayer = self.World.Players.ToDictionary(p => p, p => new DotState(p.PlayerActor.Trait<GpsWatcher>()));
 		}
 
 		public bool IsDotVisible(Player toPlayer)
 		{
-			return showToPlayer[toPlayer];
+			return stateByPlayer[toPlayer].IsVisible;
 		}
 
-		bool ShouldShowIndicator(Player toPlayer)
+		bool ShouldShowIndicator(Player toPlayer, out bool shouldRenderIndicator)
```

Comment:
What is the difference between `ShoudShowIndicator` (i.e. the return value) and `shouldRenderIndicator`?